app/src/vendor/mirador-metadata/metadataPanel.js

```diff
@@ -0,0 +1,42 @@
+(function(M) {
+
+  M.prototype.metadataPlugin = function(){
+    var mirador = this;
+
+    mirador.eventEmitter.subscribe('WINDOW_ELEMENT_UPDATED', function(event, data){
+      var windowId = data.windowId;
+
+      // get the metadata
+      var manifestUri = mirador.viewer.data[0].manifestUri;
+      var metadatas = [];
+      $.getJSON(manifestUri, function(data) {
+        metadatas = data.sequences[0].canvases;
```

Comment:
This is completely unnecessary, but if you wanted to go for a more functional style, you could transform this slightly using `map()` and `reduce()`. Again not necessary, I've just been looking for opportunities to use this in my own code lately so thought I'd suggest it:

```javascript
metadatas = data.sequences[0].canvases.reduce(function(dict, canvas) {
  dict[canvas['@id']] = canvas.metadata || [];
}, {});
```

and then when the canvas update event fires:

```javascript
var metadata = metadatas[imageId];
var html = metadata.map(function(md) {
  var value = md.value || "No value", label = md.label || "No label";
  return label + ': ' + value;
}).join("<br/>");